tutorials/getting-started/custom-widgets.md

```diff
@@ -0,0 +1,246 @@
+---
+title: "Custom widgets"
+---
+
+## Building custom widgets
+
+You've seen a lot of the widgets that come "in the box" with Apostrophe. But you can do much more by creating your own.
+
+### Custom navigation
+
+A common case: you want to build your own navigation menu. [Apostrophe's page tree is awesome](building-navigation.html) but sometimes you want to "cherrypick" pages from all over the tree, perhaps for a special footer.
+
+You could use a rich text widget and just tell users to add links manually. But they'll just break each time a page is moved around the site. And it's easier for users to mess up the formatting that way. We want something more consistent.
+
+Let's look at some custom widgets that help provide navigation. We'll start with a simple widget that adds a link in a well-formatted way.
+
+#### `link`: the simplest widget
+
+First we'll need a folder for the module:
+
+```bash
```

Comment:
Do we want to be using shell commands that might be OS-specific in our tutorial? I think making a folder can be an assumed skill, and this command would not work as expected in Windows command prompt (`-p` isn't a valid option, reversing the slashes). I think most of the commands we run will likely be OS-agnostic (i.e. anything node), so should we try to keep the entire tutorial that way?